backend/tests/unit/message_bus/operations/event/test_branch.py

```diff
@@ -84,21 +90,36 @@ async def test_merged(default_branch: Branch, init_service: InfrahubServices, pr
     mock_get_component_registry = MagicMock(return_value=mock_component_registry)
     mock_component_registry.get_component.return_value = diff_repo
 
-    with patch("infrahub.message_bus.operations.event.branch.get_component_registry", new=mock_get_component_registry):
+    with (
+        patch("infrahub.message_bus.operations.event.branch.get_component_registry", new=mock_get_component_registry),
+        patch(
+            "infrahub.services.adapters.workflow.local.WorkflowLocalExecution.submit_workflow"
+        ) as mock_submit_workflow,
+    ):
```

Comment:
Before migrating to prefect, `generate` flow was not executed during the test as `test_merged` uses a `BusRecorder` and is only meant to  test that messages are correctly sent. Now using prefect workflows, `service.workflow.submit_workflow` actually executes `generate` flow, which raises an error due to an uninitialized client. So, patching `WorkflowLocalExecution.submit_workflow` fixes this issue, and allows to make sure flows have been correctly submitted.